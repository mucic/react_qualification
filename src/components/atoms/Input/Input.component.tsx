import React from 'react'
import * as Styled from "./Input.styles";

interface InputProps {
    onChange: (value: string) => void,
    value: string;
}

function Input(props: InputProps): JSX.Element {
    const {onChange, value} = props;

    return (
        <Styled.Container>
            <input type="text" onChange={e => onChange(e.target.value)} value={value} />
        </Styled.Container>
    )
}

export default Input as React.FC<InputProps>
