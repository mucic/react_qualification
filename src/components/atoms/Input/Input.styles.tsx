import styled from "styled-components";

export const Container = styled.div`
    width: 100%;
    max-width: 400px;
    > input {
        font-family: sans-serif;
        padding: 2px;
        border-color: #789;
        border-radius: 5px;
    }
`;