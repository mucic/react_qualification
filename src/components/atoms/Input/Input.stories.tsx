import React from "react";

import { storiesOf } from "@storybook/react";

import Input from "./Input.component";

storiesOf("Input", module).add('defoult', ()=> (
    <Input onChange= {() => null} placeholder="test"/>
));