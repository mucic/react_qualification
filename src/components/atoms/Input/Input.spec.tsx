import React from "react";
import renderer from "react-test-renderer";

import Input from "./Input.component";

const inputData = {
    value: '',
    onChange: () => null
};
test("Input renders as expected", () => {
  const component = renderer.create(
    <Input {...inputData}/>
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});