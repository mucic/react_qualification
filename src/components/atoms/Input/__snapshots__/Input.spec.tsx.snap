// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders as expected 1`] = `
<div
  className="sc-AxjAm tRXsT"
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;
